### ISSUE TITLE
Type hinting error : Item "None" of "DeploymentPropertiesExtended" has no attribute "XX"

### ISSUE THREAD

**Issue by wiwski** [Azure/azure-sdk-for-python#26921]
- **Package Name**: 
azure-mgmt-resource
- **Package Version**: 
21.2.0
- **Operating System**:
Ubuntu 20.04.5 LTS
- **Python Version**: 
3.10.8

**Describe the bug**
After upgrading from version 21.1.0 to 21.2.0, type of `DeploymentExtended.properties.provisioning_state` is `None`.
Thus, running mypy give errors such as : `Item "None" of "Optional[DeploymentPropertiesExtended]" has no attribute "provisioning_state"`

**To Reproduce**
1. Upgrade to 21.2.0
2. Run `mypy` with following code :
```
poller = ResourceManagementClient(...).deployments.begin_create_or_update(...)
deployment = poller.result()
if deployment.properties.provisioning_state in (
        "Succeeded",
        "Running",
        "Ready",
    ):
        print("Deployment done !")
```

**Expected behavior**
Mypy should exit with no error.

**Additional context**
Repository and failing CI is available here : https://github.com/betagouv/euphrosyne-tools-api/pull/66

**Comment by tjprescott** [comment]
@wiwski thank you for the report. @msyyc could you please take a look?

**Comment by wiwski** [comment]
Thanks @Wzb123456789 for the workaround but the code is executed correctly. The problem is the wrong typing that triggers mypy error.
Your code is functional but it will make a second call with `deployment = ResourceManagementClient(...).deployments.get(...)` which is not necessary.
/unresolve

**Comment by Wzb123456789** [comment]
@wiwski 

I re-tried the code you used and that works fine.
![image](https://user-images.githubusercontent.com/51410816/197925933-2ca9e9eb-5361-4f46-b4e0-9d8f7808044a.png)
It is recommended that you update the `azure-mgmt-resource` to the latest version and try running your code again.

**Comment by wiwski** [comment]
@Wzb123456789 the code works fine but running `mypy .` will give an error. Have you tried running `mypy` on that code ?

**Comment by Wzb123456789** [comment]
@wiwski 

I tried mypy on the modified code, and the situation is normal.
![image](https://user-images.githubusercontent.com/51410816/198971409-96ed6793-0df0-4162-98e4-2f7ad38f1f62.png)
You can provide a screenshot of your error message and mypy version information to better solve the problem for you.

**Comment by wiwski** [comment]
Hmm, weird.. On my side I have `error: Item "None" of "Optional[DeploymentPropertiesExtended]" has no attribute "provisioning_state"`

**Comment by Wzb123456789** [comment]
Hi @wiwski 

I use the same Operating System and Python Version as you to test with no problem, this is my test code:
```python
poller = resource_client.deployments.begin_create_or_update(
    resource_group_name=GROUP_NAME,
    deployment_name=DEPLOYMENT_NAME,
    parameters={
        "properties": {
            "mode": "Incremental",
            "template": template,
            "parameters": {
                "location": {"value": "West US"}
            }
        }
    }
).result()
if poller.properties.provisioning_state in (
        "Succeeded",
        "Running",
        "Ready",
):
    print("Deployment done !")
```
Also, my python version is 3.8.4 (of course I also tested with 3.10.8), mypy: 0.982.

You can compare how your code is different.

**Comment by msyyc** [comment]
Hi @wiwski, I think you can avoid the error by making a little change like:
```
...
if deployment.properties and deployment.properties.provisioning_state:
...
```

**Comment by wiwski** [comment]
@msyyc 
Yes indeed making the change fix the issue. I thought I tried this before opening the ticket.
So sorry for wasting your time on this. Thank you for taking the time to answer and find a solution.
